code/datums/browser.dm

```diff
@@ -309,6 +309,88 @@
 	else
 		return
 
+/datum/browser/modal/preflikepicker
+	var/settings = list()
+	var/icon/preview_icon = null
+
+/datum/browser/modal/preflikepicker/New(User,Message,Title,Button1="Ok",Button2,Button3,StealFocus = 1, Timeout = FALSE,list/settings,inputtype="checkbox", width = 400, height, slidecolor)
+	if (!User)
+		return
+	src.settings = settings
+
+	..(User, ckey("[User]-[Message]-[Title]-[world.time]-[rand(1,10000)]"), Title, width, height, src, StealFocus, Timeout)
+	set_content(ShowChoices(User))
+
+/datum/browser/modal/preflikepicker/proc/ShowChoices(mob/user)
+	var/dat = ""
+
+	for (var/name in settings["mainsettings"])
+		var/setting = settings["mainsettings"][name]
+		if (setting["type"] == "datum")
+			dat += "<b>[setting["desc"]]:</b> <a href='?src=[REF(src)];setting=[name];task=input;type=datum;path=[setting["path"]]'>[setting["value"]]</a><BR>"
+		else
+			dat += "<b>[setting["desc"]]:</b> <a href='?src=[REF(src)];setting=[name];task=input;type=[setting["type"]]'>[setting["value"]]</a><BR>"
+
+	if (preview_icon)
+		dat += "<td valign='center'>"
+
+		dat += "<div class='statusDisplay'><center><img src=previewicon.png width=[preview_icon.Width()] height=[preview_icon.Height()]></center></div>"
+
+		dat += "</td>"
+
+	dat += "</tr></table>"
+
+	dat += "<hr><center><a href='?src=[REF(src)];button=1'>Ok</a> "
+
+	dat += "</center>"
+
+	return dat
+
+/datum/browser/modal/preflikepicker/Topic(href,href_list)
+	if (href_list["close"] || !user || !user.client)
+		opentime = 0
+		return
+	if (href_list["task"] == "input")
+		var/setting = href_list["setting"]
+		switch (href_list["type"])
+			if ("datum")
+				settings["mainsettings"][setting]["value"] = pick_closest_path(null, make_types_fancy(typesof(text2path(href_list["path"]))))
+			if ("string")
+				settings["mainsettings"][setting]["value"] = input(user, "Enter new value for [settings["mainsettings"][setting]["desc"]]", "Enter new value for [settings["mainsettings"][setting]["desc"]]")
+			if ("number")
+				settings["mainsettings"][setting]["value"] = input(user, "Enter new value for [settings["mainsettings"][setting]["desc"]]", "Enter new value for [settings["mainsettings"][setting]["desc"]]") as num

```

Comment:
shouldn't you use stripped_input

Reply:
not for the number thing, string maybe but... it's an admin interface, do i really care about them possibly injecting html into their own controls?

Reply:
right up until someone tries to reuse the datum for a non admin form